optuna/study/study.py

```diff
@@ -171,6 +172,81 @@ def best_trials(self) -> list[FrozenTrial]:
 
         return _get_pareto_front_trials(self)
 
+    @property
+    @experimental_func("3.6.0")
+    def best_feasible_trial(self) -> FrozenTrial:
+        """Return the best feasible trial in the study.
+        This attribute is slower than :attr:`~optuna.study.Study.best_trial`
+        because it checks for feasibility for all complete trials.
+
+        .. note::
+            This feature can only be used for single-objective optimization.
+            If your study is multi-objective,
+            use :attr:`~optuna.study.Study.best_feasible_trials` instead.
+
+        Returns:
+            A :class:`~optuna.trial.FrozenTrial` object of the best trial.
+
+        .. seealso::
+            The :ref:`reuse_best_trial` tutorial provides a detailed example of how to use this
+            method.
+
+        """
+        warnings.warn(
+            "This attribute is slower than best_trial"
+            "because it checks for feasibility for all complete trials.",
+            UserWarning,
+        )
```

Comment:
Is this warning necessary? It's already pointed out in docstring, so I don't think there's a need to display it everytime it runs. From a different perspective, following [the coding style convention](https://github.com/optuna/optuna/wiki/Coding-Style-Conventions#logging-howto), if you just want to notify the user with a warning, I think you should use `Logger.warning()`.